src/patient-registration/patient-registration.component.tsx

```diff
@@ -229,8 +320,18 @@ export const PatientRegistration: React.FC = () => {
         addresses: [addressFieldValues],
       },
     };
-    savePatient(abortController, patient)
-      .then(response => response.status == 201 && history.push(`/patient/${response.data.uuid}/chart`))
+    // handle deleted names
+    deletedNames.forEach(async name => {
+      await deletePersonName(name.nameUuid, name.personUuid, abortController);
+    });
+    // handle save patient
+    savePatient(abortController, patient, patientUuidMap['patientUuid'])
+      .then(response => {
+        if (response.ok) {
+          const url = new URLSearchParams(search).get('afterUrl') || `/patient/${response.data.uuid}/chart`;
```

Comment:
Here you here sending the user to patient chart. But this needs to be more dynamic.

Reply:
I think this is dynamic enough, we look for the defined return URL which we shall move to after the patient update is processed. This is defined by the `afterUrl` param. eg:

https://openmrs-spa.org/openmrs/spa/patient-registration/patient/959c42cd-412f-43a0-b43b-cc09a66bdc50?afterUrl=/patient/959c42cd-412f-43a0-b43b-cc09a66bdc50/chart/summary .

If the `afterUrl` parameter has a falsy value, we fall back to the patient chart.

Reply:
Yes you're right. For some reason my focus fell on the part after `||`. In fact I did not even notice the the `||`